bot/exts/fun/hangman.py

```diff
@@ -0,0 +1,140 @@
+        filtered_words = [
+            word for word in self.all_words
+            if int(min_length) < len(word) < int(max_length)
+            and int(min_unique_letters) < len(set(word)) < int(max_unique_letters)
+        ]
+
+        if not filtered_words:
+            filter_not_found_embed = Embed(
+                title=(
+                    "With the filters you provided,"
+                    "I could not find any words that fit all the filters specified.."
+                ),
+                color=Colours.soft_red,
+            )
+            await ctx.send(embed=filter_not_found_embed)
+            return
+
+        # a dictionary mapping the images of the "hung man" to the number of tries it corresponds to
+        mapping_of_images = {tries_key: image_name for tries_key, image_name in zip(range(6, -1, -1), IMAGES)}
+
+        word = choice(filtered_words)
+
+        user_guess = "_" * len(word)
+        tries = 6
+
+        hangman_embed = Embed(
+            title="Hangman",
+            color=Colours.python_blue,
+        )
+        hangman_embed.set_image(url=mapping_of_images[tries])
+        hangman_embed.add_field(
+            name=f"The word is `{user_guess}`",
+            value="Guess the word by sending a message with the letter!",
+            inline=False,
+        )
+        original_message = await ctx.send(embed=hangman_embed)
+
+        while user_guess != word:
+            try:
+                message = await self.bot.wait_for(
+                    event="message",
+                    timeout=30.0,
+                    check=lambda msg: msg.author != self.bot if singleplayer == 'm' else msg.author == ctx.author
+                )
+
+                if len(message.content) > 1:
+                    continue
+
+                elif message.content in word:
+                    positions = [idx for idx, letter in enumerate(word) if letter == message.content]
+                    user_guess = "".join(
+                        [message.content if index in positions else dash for index, dash in enumerate(user_guess)]
+                    )
+
+                elif tries - 1 <= 0:
+                    losing_embed = Embed(
+                        title="You lost.",
+                        description=f"The word was `{word}`.",
+                        color=Colours.soft_red,
+                    )
+                    await original_message.edit(embed=losing_embed)
+                    return
+
+                else:
+                    tries -= 1
+
+                hangman_embed.clear_fields()
+                hangman_embed.set_image(url=mapping_of_images[tries])
+                hangman_embed.add_field(
+                    name=f"The word is `{user_guess}`",
+                    value="Guess the word by sending a message with the letter!",
+                    inline=False,
+                )
+                await original_message.edit(embed=hangman_embed)
+            except TimeoutError:
```

Comment:
only the bot.wait_for call will raise this error, could you change this so you only surround that with a try/except, rather than the whole block?

Reply:
Yep, I'll change that